### PR TITLE
Add checkout subcommand

### PR DESCRIPTION
**Pull request by viktordick**

### CODE REVIEW

**Comment by viktordick** [comment]
I have not added tests for all the options for the new command - actually, I haven't tested all of them. Will install this locally and just try to use it in the next days/weeks.

**Review by jan-jockusch — DISMISSED** [review]
LGTM!

**Comment by viktordick** [comment]
It seems this does not yet work with the `--reset` parameter

```
[I] 11:19:25 vdick@vdick-notebook ~/g/dbutils-zoperepo> zodbsync checkout -b 242827-pcordtype-cols --reset origin/master
2023-06-15T11:19:44.351 ZODBSync INFO Checking out 242827-pcordtype-cols.
Switched to a new branch '242827-pcordtype-cols'
2023-06-15T11:19:44.354 ZODBSync ERROR Error during operation. Resetting.
HEAD is now at 9b853f11f8c T241935: Add documentation
Switched to branch '241935-comacct-pdf'
Your branch is up to date with 'origin/241935-comacct-pdf'.
Traceback (most recent call last):
  File "/usr/share/perfact/zope4/bin/zodbsync", line 6, in <module>
    Runner().run()
  File "/usr/share/perfact/zope4/lib/python3.10/site-packages/perfact/zodbsync/main.py", line 136, in run
    self.parse(*argv).run()
  File "/usr/share/perfact/zope4/lib/python3.10/site-packages/perfact/zodbsync/subcommand.py", line 53, in wrapper
    result = func(self, *args, **kwargs)
  File "/usr/share/perfact/zope4/lib/python3.10/site-packages/perfact/zodbsync/subcommand.py", line 159, in wrapper
    func(self, *args, **kwargs)
  File "/usr/share/perfact/zope4/lib/python3.10/site-packages/perfact/zodbsync/commands/checkout.py", line 49, in run
    self.gitcmd_run(['reset', '--hard', self.args.reset])
  File "/usr/share/perfact/zope4/lib/python3.10/site-packages/perfact/zodbsync/subcommand.py", line 67, in gitcmd_run
    subprocess.check_call(self.gitcmd(*args))
  File "/usr/lib/python3.10/subprocess.py", line 364, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/usr/lib/python3.10/subprocess.py", line 345, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.10/subprocess.py", line 969, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.10/subprocess.py", line 1778, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
TypeError: expected str, bytes or os.PathLike object, not list
Traceback (most recent call last):
  File "/usr/bin/zodbsync", line 15, in <module>
    subprocess.run(cmd, check=True)
  File "/usr/lib/python3.10/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/share/perfact/zope4/bin/zodbsync', '-c', '/etc/perfact/modsync/zodb3.py', 'checkout', '-b', '242827-pcordtype-cols', '--reset', 'origin/master']' returned non-zero exit status 1.
```

File: perfact/zodbsync/commands/checkout.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        self.logger.info('Checking out %s.' % self.args.branch)
+        cmd = ['checkout']
+        if self.args.b:
+            cmd.append('-b')
+        cmd.append(self.args.branch)
+        if self.args.b and self.args.track:
+            cmd.extend(['--track', self.args.track])
+        self.gitcmd_run(*cmd)
+        if self.args.reset:
+            self.gitcmd_run(['reset', '--hard', self.args.reset])
```
Comment (Wiseqube) [inline perfact/zodbsync/commands/checkout.py:49]: I think for both this as well as the rebase call the list needs to be unpacked

Comment (viktordick) [reply perfact/zodbsync/commands/checkout.py]: Thanks, that was it.

**Review by Wiseqube — APPROVED** [review]
LGTM

**Pull request merged**